File: src/main/java/com/ddd/trainings/proposal/domain/TemplateId.java
package com.ddd.trainings.proposal.domain;

import com.ddd.utils.BaseId;
import java.util.UUID;
import lombok.EqualsAndHashCode;
import lombok.Getter;

@EqualsAndHashCode(of = "value", callSuper = false)
public class TemplateId extends BaseId {

  @Getter
  private final String value;

  public TemplateId(String value) {
    this.value = value;
  }

  public static TemplateId create() {
    return new TemplateId(UUID.randomUUID().toString());
  }
} 

File: src/Ddd/Trainings/Proposal/Domain/TemplateId.cs
using System;
using Ddd.Utils;

namespace Ddd.Trainings.Proposal.Domain
{
    public class TemplateId(string value) : BaseId
    {
    private readonly string _value = value;

    public static TemplateId Create()
        {
            return new TemplateId(Guid.NewGuid().ToString());
        }

    public override bool Equals(object? obj)
    {
      return obj is TemplateId id &&
             _value == id._value;
    }

    public override int GetHashCode()
    {
      return HashCode.Combine(_value);
    }

    public override string GetValue()
    {
      return _value;
    }
  }
} 